makefiles/tools/serial.inc.mk

```
@@ -53,7 +53,13 @@ else ifeq ($(RIOT_TERMINAL),picocom)
   TERMPROG  ?= picocom
   TERMFLAGS ?= --nolock --imap lfcrlf --baud "$(BAUD)" "$(PORT)"
 else ifeq ($(RIOT_TERMINAL),miniterm)
-  TERMPROG  ?= miniterm.py
+  # Check if miniterm.py is available in the path, if not use just miniterm
+  # since new versions will only have miniterm and not miniterm.py
+  ifeq (, $(shell which miniterm.py))
+    TERMPROG ?= miniterm
+  else
+    TERMPROG ?= miniterm.py
+  endif
```

```suggestion
  # Check if miniterm.py is available in the path, if not use just miniterm
  # since new versions will only have miniterm and not miniterm.py
  TERMPROG ?= $(shell command -v miniterm.py 2>/dev/null)
  ifeq (,$(TERMPROG))
    TERMPROG ?= miniterm
  endif
```

If I recall correctly there were some issues with GNU make emulating `$(shell which)` rather than actually calling `which`, but failing to do so correctly on some versions. `command -v` was the portable option.

Beware: Suggestion is untested

---

Great, I will have to try it tomorrow... No board and no time.  It looks like @jia200x didn't trigger the release yet so maybe I do that the hard feature freeze tomorrow.

---

Didn't work out-of-the-box but I used the more portable solution.